### PR TITLE
Don't reuse packet ids.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: tests/packet_test_client.toit [hunk 1]
```diff
@@ -77,7 +77,7 @@ with_packet_client create_transport/Lambda [block]
 
   client.when_running:
     client.subscribe idle_topic --max_qos=0
-    wait_for_idle.call
+    // wait_for_idle.call
```
Comment (kasperl) [inline tests/packet_test_client.toit:80]: In or out?

File: src/full_client.toit [hunk 2]
```diff
@@ -416,7 +416,11 @@ class Session_:
     persistence_store_.remove_persisted_with_id id
 
   next_packet_id -> int:
-    return next_packet_id_++
+    while true:
+      candidate := next_packet_id_
+      next_packet_id_ = (next_packet_id_ + 1) & 0xFFFF
```
Comment (kasperl) [inline src/full_client.toit:421]: Is the id always sent as a 16 bit value? Maybe use a named constant for the mask.

File: src/full_client.toit [hunk 3]
```diff
@@ -416,7 +416,11 @@ class Session_:
...
+      // Only use the candidate if we are not waiting for an ACK of that id.
+      if not persistence_store_.get candidate: return candidate
```
Comment (kasperl) [inline src/full_client.toit:423]: I might flip it around and make the comment simpler: 

```
// If we're waiting for an ACK of the candidate id, we pick another.
if persistence_store_.get candidate: continue
return candidate
```

File: src/full_client.toit (hunk 2 above)
Comment (kasperl) [inline src/full_client.toit:421]:
```suggestion
      next_packet_id_ = (candidate + 1) & 0xFFFF
```

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tests/packet_test_client.toit (hunk 1 above)
Comment (floitsch) [reply tests/packet_test_client.toit]: out.
done.

File: src/full_client.toit (hunk 2 above)
Comment (floitsch) [reply src/full_client.toit]: done.

Comment (floitsch) [reply src/full_client.toit]: done.

File: src/full_client.toit (hunk 3 above)
Comment (floitsch) [reply src/full_client.toit]: done.

**Pull request merged**